### ISSUE TITLE
Crash while running on synology NAS

### ISSUE THREAD

**Issue by DarthLowen** [theBlackDragon/conan-exiles-dedicated-server#1]
Hi Bert,

I successfully created an image and container based on your dockerfile, awesome work! The container runs on my virtual PC (ubuntu 20.04) but crashes on my synology NAS (same image saved on ubuntu and loaded on synology) Any idea why this might be happening?

I attached the log file (unintuitively newest on top): [conanexile.html.pdf](https://github.com/theBlackDragon/conan-exiles-dedicated-server/files/6870889/conanexile.html.pdf)

All help much appreciated!

Groeten uit Ronse (ik maak deze image trouwens voor mijn maat in Ouwegem, small world! ;-) )

Nic.

**Comment by theBlackDragon** [comment]
Hi Nic,

Small world indeed!

Is the NAS x86, Or is it ARM? If the latter that could be the issue as the steamcmd container uses i386 architecture for wine(since there's no native Linux version of the CE server software we have to resort to using Wine)

There is a Wine ARM port, but as far as I've been able to find it would only be able to run Windows software already compiled for ARM, meaning it wouldn't work for Conan.

If it _is_ x86 then we'll have to dig deeper...

**Comment by DarthLowen** [comment]
It's a DS218+ with Intel Celeron J3355, 2 core @2GHz and 2GB RAM.

The container created from the image by [alinmear](https://hub.docker.com/r/alinmear/docker-conanexiles/) stays running, however the server exits every 5 minutes with status 5, which could be due to bad configuration, haven't really taken the time to figure out how to configure that installation decently.

Thank god it's not an ARM cpu, qemu and wine combined is a bridge too far...if you're interested, it's called [hangover](https://github.com/AndreRH/hangover)

**Comment by theBlackDragon** [comment]
> It's a DS218+ with Intel Celeron J3355, 2 core @2GHz and 2GB RAM.

The game server likes to eat up RAM, it also like to crash due to running out of memory (server isn't the best written piece of software and has quite a few memory leaks. Mods tend to exacerbate the issue)

I recreated and restarted my container to verify whether any Debian updates would have messed things up, and I notice that a fresh instance is already eating up 6.5GB of memory. I have no idea how well (or badly) it will handle more memory constrained systems, but it seems 8GB is the [minimum requirement](https://conanexiles.fandom.com/wiki/Dedicated_server_system_requirements), so that could potentially explain the issue with the alinmear container. If it runs it will probably have to dip quite deeply into swap space, which might result in some pretty bad performance.

Of course, doesn't necessarily explain why this image just doesn't want to even start.
> 
> The container created from the image by [alinmear](https://hub.docker.com/r/alinmear/docker-conanexiles/) stays running, however the server exits every 5 minutes with status 5, which could be due to bad configuration, haven't really taken the time to figure out how to configure that installation decently.

Aside from the additional features/complexity two key differences are that I used Debian as a base, rather than Ubuntu (leads to lighter images, but was a bit harder to deal with Wine initially) and that I purposefully do not run the game as root (I notice they've been trying to fix that for a while now). 

Just noticed I didn't implement the option to specify the UID and GID for the steam user on the container (linuxserver.io style), which could potentially lead to permissions issues when they don't match anything (or something unexpected) on the host, something I should fix.

> Thank god it's not an ARM cpu, qemu and wine combined is a bridge too far...if you're interested, it's called [hangover](https://github.com/AndreRH/hangover)

The naming certainly seems appropriate... ;)
